<launch>

    <arg name="raw_cloud"         default="/os_cloud_node/points" />
    <arg name="filtered_cloud"    default="/cloud_filtered" />
    <arg name="predicted_pose"    default="/odometry_node/odometry_estimate" />
    <arg name="odom_frame"        default="odom" />
    <arg name="epsilon"           default="0.84" />

    <arg name="model_weights_pth" default="/sps/best_models/420_601.ckpt" />

    <node pkg="sps_filter" type="sps_node.py" name="Stable_Points_Segmentation" output="screen">
        <param name="raw_cloud"         type="string" value="$(arg raw_cloud)"/>
        <param name="filtered_cloud"    type="string" value="$(arg filtered_cloud)"/>
        <param name="predicted_pose"    type="string" value="$(arg predicted_pose)"/>
        <param name="odom_frame"        type="string" value="$(arg odom_frame)"/>
        <param name="model_weights_pth" type="string" value="$(arg model_weights_pth)"/>
        <param name="epsilon"           type="double" value="$(arg epsilon)"/>
    </node>

</launch>
